internal/encryption/base.go

```diff
@@ -229,12 +230,8 @@ func (base *baseEncryption) decrypt(data []byte, validator func([]byte) error) (
 		errs = append(errs, fmt.Errorf("attempted decryption failed for %s: %w", base.name, err))
 	}
 
-	// This is good enough for now until we have better/distinct errors
-	errMessage := "decryption failed for all provided methods: "
-	sep := ""
-	for _, err := range errs {
-		errMessage += err.Error() + sep
-		sep = "\n"
-	}
+	errs = append([]error{fmt.Errorf("decryption failed for all provided methods")}, errs...)
+	errMessage := errors.Join(errs...).Error()
```

Comment:
We should probably just return the result of errors.Join() here instead of creating a new error below.